### PR TITLE
Install terraform in the vm script

### PR DESCRIPTION
**Pull request by DevinBeu**
Run this script after it has been moved into the shell to install terraform and dependencies.

### CODE REVIEW

File: installTerraInVm.sh [hunk 1]
```diff
@@ -0,0 +1,53 @@
+# first make sure all terraform dependencies are installed
+# Check if gnupg is installed
+if which gpg > /dev/null 2>&1; then
+  echo 'gnupg and software-properties-common are already installed'
+else
+  echo 'Installing gnupg and software-properties-common...'
+  sudo apt-get update && sudo apt-get install -y gnupg software-properties-common
```
Comment (stephenlauck) [inline installTerraInVm.sh]: you can create a separate check using `apt-cache show gpg` to know whether you need to run apt-update or not.

File: installTerraInVm.sh [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+  gpg --dearmor | \
+  sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg > /dev/null
+fi
+
+# Verify the key's fingerprint
+if gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint | grep -q 'HashiCorp'; then
+  echo 'HashiCorp GPG key fingerprint verified.'
+else
+  echo 'Error: HashiCorp GPG key fingerprint verification failed.'
+  exit 1
```
Comment (stephenlauck) [inline installTerraInVm.sh]: All conditional blocks should actually fix the issue to desired state not FAIL. What would be the fix here? What are you trying to do?

File: installTerraInVm.sh [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+else
+  echo 'Adding HashiCorp repository.'
+  echo "deb [signed-by=/usr/share/keyrings/hashicorp-archive-keyring.gpg] \
+https://apt.releases.hashicorp.com $(lsb_release -cs) main" | \
+  sudo tee /etc/apt/sources.list.d/hashicorp.list
+fi
+
+# Update package information from HashiCorp
+echo 'Updating package information from HashiCorp repository.'
+sudo apt update
```
Comment (stephenlauck) [inline installTerraInVm.sh]: Why no conditional here? What is special about apt update that you don't need to put it in a conditional?

File: installTerraInVm.sh [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+  echo 'HashiCorp GPG key is already present.'
+else
+  echo 'HashiCorp GPG key not found. Adding the key now.'
+  wget -O- https://apt.releases.hashicorp.com/gpg | \
+  gpg --dearmor | \
+  sudo tee /usr/share/keyrings/hashicorp-archive-keyring.gpg > /dev/null
+fi
+
+# Verify the key's fingerprint
+if gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint | grep -q 'HashiCorp'; then
```
Comment (stephenlauck) [inline installTerraInVm.sh]: Maybe doing this check is a good way to check if the key is present? You can do 2 things at once by moving this check to be the check on line 11.

**Review by stephenlauck — CHANGES_REQUESTED** [review]

File: installTerraInVm.sh (hunk 1 above)
Comment (DevinBeu) [reply installTerraInVm.sh]: The new file is under the commits tab, i dont know how to relink it?

File: installTerraInVm.sh [hunk 5]
```diff
@@ -1,53 +1,54 @@
...
+    sudo apt-get update
+    echo 'Installing gnupg and software-properties-common...'
+    sudo apt-get install -y gnupg software-properties-common
 fi
 
-# Check if the HashiCorp GPG key is present
-if gpg --no-default-keyring --keyring /usr/share/keyrings/hashicorp-archive-keyring.gpg --fingerprint | grep -q 'HashiCorp'; then
-  echo 'HashiCorp GPG key is already present.'
+# Check for apt-cache status
+if apt-cache search gnupg && apt-cache search software-properties-common; then
```
Comment (stephenlauck) [inline installTerraInVm.sh]: `apt-cache show gnupg` would be good enough. Be careful putting logical AND in your tests (desired state)

File: installTerraInVm.sh [hunk 6]
```diff
@@ -1,53 +1,54 @@
...
 https://apt.releases.hashicorp.com $(lsb_release -cs) main" | \
-  sudo tee /etc/apt/sources.list.d/hashicorp.list
+    sudo tee /etc/apt/sources.list.d/hashicorp.list > /dev/null
 fi
 
-# Update package information from HashiCorp
-echo 'Updating package information from HashiCorp repository.'
-sudo apt update
+# Check if apt needs updating after potential repository addition
+if test -f /etc/apt/sources.list.d/hashicorp.list && find /etc/apt/sources.list.d/hashicorp.list -mmin -5; then
```
Comment (stephenlauck) [inline installTerraInVm.sh]: The check for terraform to be in the apt-cache would be good enough you don't need to search in the sources and this doesn't work anyway.

You can just throw another check like:

```
if (apt-cache show terraform)
then
...
else
  apt update
fi
```

**Review by stephenlauck — CHANGES_REQUESTED** [review]

File: installTerraInVm.sh [hunk 7]
```diff
@@ -1,7 +1,7 @@
 #!/bin/bash
 
 # Check if gnupg is installed
-if gpg --version; then
+if (apt-cache show gnupg); then
     echo 'gnupg is already installed'
```
Comment (stephenlauck) [inline installTerraInVm.sh]: It's not installed, it's ready to be installed or it's "in the apt cache".

File: installTerraInVm.sh [hunk 8]
```diff
@@ -38,9 +38,11 @@ https://apt.releases.hashicorp.com $(lsb_release -cs) main" | \
 fi
 
 # Check if apt needs updating after potential repository addition
-if test -f /etc/apt/sources.list.d/hashicorp.list && find /etc/apt/sources.list.d/hashicorp.list -mmin -5; then
-    echo 'Running apt update due to recent repository changes...'
-    sudo apt update
+if (apt-cache show terraform)
+  then 
+  echo 'terrform package is already up to date'
```
Comment (stephenlauck) [inline installTerraInVm.sh]: the message should be something like "terraform is in the apt cache and available to be installed"

**Review by stephenlauck — CHANGES_REQUESTED** [review]

File: installTerraInVm.sh [hunk 9]
```diff
@@ -0,0 +1,49 @@
+
+#!/bin/bash
+
+# Check if gnupg are in apt cache-if not update cache and install dependencies
+if (apt-cache show gnupg); then
+    echo 'gnupg is already in the apt-cache'
+else
+    sudo apt-get update
+    echo 'Installing gnupg and software-properties-common...'
+    sudo apt-get install -y gnupg software-properties-common
```
Comment (stephenlauck) [inline installTerraInVm.sh:10]: Might be better to have 2 explicit checks, one for "X shoudl be installed" and one for "X should be in the apt cache" as they are 2 different things.

**Review by stephenlauck — APPROVED** [review]

**Pull request closed without merging**